lib/core/src/Cardano/Wallet.hs

```diff
@@ -1264,6 +1272,139 @@ normalizeDelegationAddress s addr = do
                                   Transaction
 -------------------------------------------------------------------------------}
 
+balanceTransaction
+    :: forall m s k. (GenChange s, MonadRandom m)
+    => Tracer m WalletLog
+    -> ArgGenChange s
+    -> (W.ProtocolParameters, Cardano.ProtocolParameters)
+    -> (UTxOIndex, Wallet s, Set Tx)
+    -> TransactionLayer k SealedTx
+    -> [(TxIn, TxOut)] -- ^ Resolved inputs
+    -> SealedTx
+    -> ExceptT ErrBalanceTx m SealedTx
+balanceTransaction
+    tr
+    generateChange
+    (pp, nodePParams)
+    (internalUtxoAvailable, wallet, pendingTxs)
+    tl
+    resolvedInputs
+    partialTx
+    = do
```

Comment:
There are quite a lot of parameters! Would it be helpful to define a record type to hold them all?

This might also make it clearer at call sites which parameter is which. The errors when you accidentally omit a parameter can be more concise, too.